[Airiinnn] iP

Airiinnn opened the pull request:
# Kat Bot

> Meow ([Source](https://dictionary.cambridge.org/us/dictionary/english/meow))

Kat keeps track of your
1. Todos
1. Deadlines
1. Events

It is
* CLI first
* Easy to learn
* **FAST** to use 🔥 

Features:
- [x] CLI
- [ ] GUI

___

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```Java
public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    TaskStorage taskStorage = new TaskStorageImpl();
    MessageParser messageParser = new MessageParserImpl(taskStorage);
    ChatBot chatBot = new KatChatBotImpl(scanner, messageParser);

    chatBot.start();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Airiinnn committed:
Rename Duke to Kat

Airiinnn committed:
Implement initial greet and exit

Airiinnn committed:
Support echoing responses

Airiinnn committed:
Implement task list

Airiinnn committed:
Support done status in tasks

Airiinnn committed:
Support different types of tasks

Airiinnn committed:
Fix indentation according to style guide

Airiinnn committed:
Add text ui test

Airiinnn committed:
Refactor

Airiinnn committed:
Add exception handling

Airiinnn committed:
Add tests for exception handling

Airiinnn committed:
Rename name field to description in tasks

Airiinnn committed:
Support deleting of tasks

Airiinnn committed:
Add tests for deleting

Airiinnn committed:
Add task serialization

Airiinnn committed:
Add task deserialization

Airiinnn committed:
Cleanup

Airiinnn committed:
Merge branch 'branch-Level-7'

Airiinnn committed:
Change dates to LocalDate

Airiinnn committed:
Merge branch 'branch-Level-8'

Airiinnn committed:
Merge branch 'add-gradle-support'

Airiinnn committed:
Update build.gradle

Airiinnn committed:
Add unit tests for MessageParserImpl

Airiinnn committed:
Refactor to use dependency injection

Airiinnn committed:
Add unit tests for AbstractTask

Airiinnn committed:
Merge branch 'branch-A-JUnit'

Airiinnn committed:
Add JavaDoc comments for interfaces

Airiinnn committed:
Add JavaDoc comments for implementations

Airiinnn committed:
Reorder imports

Airiinnn committed:
Add find functionality

Airiinnn committed:
Add unit test for find command parsing

Airiinnn committed:
Merge branch 'branch-A-JavaDoc'

Airiinnn committed:
Merge branch 'branch-A-CodingStandard'

Airiinnn committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/chatbot/TaskStorage.java
#	src/test/java/chatbot/impl/MessageParserImplTest.java

LeeZeHao commented on src/main/java/chatbot/Main.java:
```diff
@@ -0,0 +1,20 @@
+import chatbot.impl.KatChatBotImpl;
+import chatbot.impl.MessageParserImpl;
+import chatbot.impl.TaskStorageImpl;
+
```
Shouldn't there still be a JavaDoc here, even though it is the Main class?

LeeZeHao commented on src/main/java/chatbot/impl/MessageParserImpl.java:
```diff
@@ -0,0 +1,215 @@
+
+    /**
+     * Handles the 'list' command.
+     *
```
Although JavaDoc for private methods was not mentioned in requirements, it's great that you've done it! Improves readability a lot

LeeZeHao commented on src/main/java/chatbot/impl/MessageParserImpl.java:
```diff
@@ -0,0 +1,215 @@
+/**
+ * Implements the MessageParser interface to handle various types of user messages.
+ */
+public class MessageParserImpl implements MessageParser {
```
Would it be better to have named the interface MessageParserInterface (or similar) so that the class itself could just be named as MessageParser?

LeeZeHao commented on src/main/java/chatbot/impl/TaskStorageImpl.java:
```diff
@@ -0,0 +1,142 @@
+/**
+ * Implements the TaskStorage interface to manage tasks in a custom file-based storage system.
+ */
+public class TaskStorageImpl implements TaskStorage {
```
Same as above, would it be better to have named the interface TaskStorageInterface(or similar) so that the class itself could just be named as TaskStorage?

LeeZeHao commented on src/main/java/chatbot/impl/TaskStorageImpl.java:
```diff
@@ -0,0 +1,142 @@
+    }
+
+    /**
+     * Returns a string representation of all tasks in the storage.
```
Seems like this JavaDoc comment is in front of the wrong method (should be in front of toString()?)

LeeZeHao commented on src/main/java/chatbot/impl/tasks/TodoTask.java:
```diff
@@ -0,0 +1,48 @@
+        return todoTask;
+    }
+
+    // Todo: Handle case where '|' is part of description
```
Do add JavaDoc comment whenever ready!

LeeZeHao reviewed (commented):
Generally follows coding standard (especially naming-wise) quite well!
Only a few JavaDoc documentation issues to clean up!

Airiinnn committed:
Add checkstyle

Airiinnn committed:
Clean up

Airiinnn committed:
Remove deprecated text-ui-test

Airiinnn committed:
Add GUI

Airiinnn committed:
Merge branch 'branch-Level-10'

Airiinnn committed:
Add assertions to MessageParserImpl for improved robustness

The MessageParserImpl class currently lacks explicit checks for its
assumptions about input and state. This can lead to subtle bugs that
are difficult to detect and debug.

We need to make these assumptions explicit to catch potential issues
early in the development process and improve code reliability.

Add assertions throughout MessageParserImpl to validate inputs, check
state, and verify assumptions.

This approach allows us to catch programming errors during development
and testing without affecting runtime behavior in production, as
assertions are typically disabled in production environments.

These changes complement, rather than replace, the existing runtime
error handling, providing an additional layer of safety during the
development process.

Airiinnn committed:
Merge pull request #2 from Airiinnn/branch-A-Assertions

Add assertions to MessageParserImpl for improved robustness

Airiinnn committed:
Refactor magic literals and remove dead code

The project currently contains several magic literals some unused code
sections. This reduces code readability and maintainability, making it
harder for developers to understand and modify the class.

We need to improve the code quality to make it more self-documenting
and easier to maintain in the long term.

Extract magic literals into named constants and remove unused code
sections from KatChatBotImpl and Main.

This approach enhances code readability by giving meaningful names to
values that were previously unexplained numbers or strings. It also
simplifies the codebase by removing unnecessary code that was adding
cognitive overhead without providing any functionality.

These changes set the stage for easier future modifications and reduce
the risk of introducing bugs when maintaining or extending the class.

Airiinnn committed:
Clean up and remove unnecessary methods and fields

Airiinnn committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/chatbot/impl/MessageParserImpl.java

Airiinnn committed:
Merge pull request #3 from Airiinnn/branch-A-CodeQuality

Branch a code quality

Airiinnn committed:
Add help command (C-Help)

Airiinnn committed:
Merge branch 'branch-BCD-Extension'

Airiinnn committed:
Add user and chatbot profile pictures

Airiinnn committed:
Add welcome message in GUI on startup

Airiinnn committed:
Add support for resizing GUI window

Airiinnn committed:
Merge branch 'branch-A-BetterGui'

Airiinnn committed:
Update assertions for appropriate usage

Airiinnn committed:
Add support for terminating session with command

Airiinnn committed:
Clean up and remove dead code

Airiinnn committed:
Add AI usage to README.md

Airiinnn committed:
Refactor for better project structure

Airiinnn committed:
Add user guide

Airiinnn committed:
Update user guide

Airiinnn committed:
Fix gradle configuration
